python/pyarrow/pandas_compat.py

```diff
@@ -544,9 +547,14 @@ def table_to_blockmanager(options, table, memory_pool, nthreads=1,
 
     column_strings = [x.name for x in block_table.itercolumns()]
     if columns:
-        columns_name_dict = {
-            c.get('field_name', str(c['name'])): c['name'] for c in columns
-        }
+        columns_name_dict = {}
+        for c in columns:
+            column_name = c['name']
+            if not isinstance(column_name, six.text_type):
+                column_name = str(column_name)
```

Comment:
This should also be `six.text_type` so that we get `unicode` in Python 2. Probably, using `.decode('utf-8')` might also be the better option.

Reply:
There is also `frombytes` in `pyarrow.compat`